lib/jobs/switch-node-relations.js

```diff
@@ -0,0 +1,212 @@
+        })
+        .then(function () {
+            self._done();
+        })
+        .catch(function (err) {
+            self._done(err);
+        });
+
+    };
+
+    /**
+     * search in snmp-1 catalog data for switch mac addresses
+     * returns an array of switch mac addresses
+     */
+    SwitchRelationsJob.prototype._getMacAddressList = function getMacAddressList() {
+        var self = this;
+        var data;
+        var macArr = [];
+        return waterline.catalogs.findMostRecent({
+            node: self.nodeId,
+            source: 'snmp-1'
+        })
+        .then(function (catalog) {
+            data = catalog.data;
+            return Promise.resolve(Object.keys(catalog.data));
+        })
+        .filter(function (key) {
+            return /ifPhysAddress_/.test(key);
+        })
+        .each(function (phys) {
+            if (_validateMacStrFormat(data[phys])) {
+                macArr.push(_adjustMacStrFormat(data[phys]));
+            }
+        })
+        .then(function () {
+            if (macArr.length === 0) {
+                throw new Error('couldnt find ports mac addresses in catalog');
+            }
+            return Promise.resolve(macArr);
+        });
+    };
+
+    /**
+     * update an existing entry in nodes relations or
+     * create one if it doesn't exist
+     * return a promise
+     */
+    SwitchRelationsJob.prototype._updateSwitchRelations =
+        function updateSwitchRelations(lldpData, info) {
+            var self = this;
+            return waterline.nodes.findByIdentifier(lldpData.node)
+            .then(function (node) {
+                if (!node) {
+                    return Promise.reject('Could not find node with identifier ' + lldpData.node);
+                }
+                else {
+                    return Promise.resolve(node.relations);
+                }
+            })
+            .filter(function (entry) {
+                return (entry.targets.indexOf(self.nodeId) < 0);
+            })
+            .then(function (relations) {
+                relations.push({
+                    relationType: RELATION_TYPE,
+                    targets: [self.nodeId],
+                    info: info
+                });
+                return waterline.nodes.updateByIdentifier(
+                    lldpData.node,
+                    { relations: relations });
+            });
+
+        };
+
+    function _getCatalogsBySource(src) {
+        return waterline.catalogs.find({
+            source: src
+        });
+    }
+
+    /**
+     * Iterate through switch mac addresses list
+     * for a match in nodes lldp catalog
+     * return a promise
+     */
+    function _findMatchingMac(nodeLldpData, lldpMacList) {
+        var relationsList = {};
+        return Promise.all(
+        _.forEach(_.keys(nodeLldpData.data), function (ethPort) {
+            //validate mac first
+            var lldpMac = catalogSearch.getPath(nodeLldpData.data[ethPort], 'chassis.mac');
```

Comment:
like here's the lldp data from a virtual node catalog: 
```
 {
    "node": "5761ca481b0388626870dddb",
    "source": "lldp",
    "data": {
      "eth0": {
        "via": "LLDP",
        "rid": "2",
        "age": "0 day, 00:00:11",
        "chassis": {
          "mac": "08:00:27:b1:13:43",
          "name": "my-arista",
          "descr": "Arista Networks EOS version 4.13.7M running on an Arista Networks vEOS",
          "mgmt-ip": "172.31.128.36",
          "Bridge": {
            "enabled": "on"
          },
          "Router": {
            "enabled": "off"
          }
        },
        "port": {
          "ifname": "Management1",
          "descr": "Not received"
        }
      }
    }
```
and here's some of the pertinent snmp data from the virtual switch:

```
"LLDP-MIB::lldpLocChassisIdSubtype_0": "macAddress",
"LLDP-MIB::lldpLocChassisId_0": "08 00 27 B1 13 43 "

"IF-MIB::ifPhysAddress_1": "8:0:27:aa:c8:8e",
"IF-MIB::ifPhysAddress_2": "8:0:27:ca:c6:a",
"IF-MIB::ifPhysAddress_999001": "8:0:27:1e:27:4e"
```